addons/mail/static/src/messaging/entity/thread_viewer/thread_viewer.js

```diff
@@ -191,14 +221,44 @@ function ThreadViewerFactory({ Entity }) {
             });
 
             if (thread && this.thread !== thread) {
+                if (this._loaderTimeout) {
+                    clearTimeout(this._loaderTimeout);
+                }
                 this.link({ thread });
                 if (!this.threadCache.isLoaded && !this.threadCache.isLoading) {
                     this.threadCache.loadMessages();
                 }
             }
             if (this.threadCache !== prevThreadCache) {
+                if (this._loaderTimeout) {
+                    clearTimeout(this._loaderTimeout);
+                }
                 this._writeComponentHint('change-of-thread-cache');
             }
+
+            if(!this.threadCache) {
+                this._write({
+                    hasLoader:false,
+                    _loaderTimeout: null
+                });
+            }
+            else {
+                if (this.threadCache.isLoading || !this.threadCache.isLoaded) {
```

Comment:
should be just `isLoading`

Reply:
Just took the condition that was in the template earlier, I indeed didn't understand why both were checked so ok I'll change